### PR TITLE
Add missing types in Core

### PR DESCRIPTION
**Pull request by stof**
This is part of my work to increase the phpstan level. At level 6, phpstan wants all properties and methods to define their types. This does the work in the Core package (excluding the generated Sts client).

### CODE REVIEW

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: src/Core/src/Request.php [hunk 1]
```diff
@@ -90,23 +111,23 @@ public function getBody(): RequestStream
...
 
-    public function removeQueryAttribute($name): void
+    public function removeQueryAttribute(string $name): void
     {
         unset($this->query[$name]);
         $this->endpoint = '';
     }
 
-    public function setQueryAttribute($name, $value): void
+    public function setQueryAttribute(string $name, string $value): void
```
[the remark this answers is not in the document]
Comment (stof) [reply src/Core/src/Request.php:130]: this class _is_ internal

**Review by jderusse — APPROVED** [review]

File: src/Core/src/Result.php [hunk 2]
```diff
@@ -25,16 +25,22 @@ class Result
...
+     */
     private $response;
 
     /**
      * @var self[]
      */
     private $prefetchResults = [];
 
-    public function __construct(Response $response, ?AbstractApi $awsClient = null, $request = null)
+    public function __construct(Response $response, ?AbstractApi $awsClient = null, ?object $request = null)
```
[the remark this answers is not in the document]
Comment (stof) [reply src/Core/src/Result.php:43]: passing anything else than an object or null would already lead to an exception later (in the generated code of child classes) due to invalid usages.
Thus, this constructor is used by the generated code which passes the input object there.

File: src/Core/src/Request.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (GrahamCampbell) [reply src/Core/src/Request.php:130]: It is not marked as `@internal`.

[the remark this answers is not in the document]
Comment (jderusse) [reply src/Core/src/Request.php:130]: the class is marked `@internal` :thinking: 

https://github.com/async-aws/aws/blob/8c3a8e7c1fbd54b5aabfc84eefdd7387cb5014ba/src/Core/src/Request.php#L13

[the remark this answers is not in the document]
Comment (GrahamCampbell) [reply src/Core/src/Request.php:130]: oh, hahaha. i missed that.

**Review by GrahamCampbell — APPROVED** [review]
:shipit:

**Pull request merged**